bus/manager.go

```diff
@@ -0,0 +1,109 @@
+package bus
+
+import (
+	"os"
+
+	"github.com/kairos-io/kairos-sdk/types"
+	"github.com/mudler/go-pluggable"
+)
+
+func NewBus(withEvents ...pluggable.EventType) *Bus {
+	if len(withEvents) == 0 {
+		withEvents = AllEvents
+	}
+	return &Bus{
+		Manager: pluggable.NewManager(withEvents),
+	}
+}
+
+type Bus struct {
+	*pluggable.Manager
+	registered     bool
+	logger         *types.KairosLogger // Fully override the logger
+	logLevel       string              // Log level for the logger, defaults to "info" unless BUS_DEBUG is set to "true". This only valid if logger is not set.
+	logName        string              // Name of the logger, defaults to "bus". This only valid if logger is not set.
+	providerPrefix string              // Prefix for provider plugins, defaults to "agent-provider". This is used to autoload providers.
+	providerPaths  []string            // Paths to search for provider plugins, defaults to system and current working directory.
+}
+
+func (b *Bus) LoadProviders() {
+	b.Autoload(b.providerPrefix, b.providerPaths...).Register()
```

Comment:
not sure if we want to return the Manager here that is returned by `Register`, but might be handy in cases the caller might need it?